astropy/coordinates/tests/test_sky_coord.py

```diff
@@ -1556,3 +1557,23 @@ class MockHeliographicStonyhurst(BaseCoordinateFrame):
 
     fr = MockHeliographicStonyhurst(lon=1*u.deg, lat=2*u.deg, radius=10*u.au)
     SkyCoord(0*u.deg, fr.lat, fr.radius, frame=fr) # this was the failure
+
+
+def test_represent_as_dict_proper_motion():
+
+    c = SkyCoord(ra=1*u.deg, dec=2*u.deg, pm_ra_cosdec=2*u.mas/u.yr,
+                 pm_dec=1*u.mas/u.yr)
+
+    newc = SkyCoord.info._construct_from_dict(c.info._represent_as_dict())
+
+    assert skycoord_equal(c, newc)
```

Comment:
I think `skycoord_equal` does not actually check differentials either...

Reply:
Hmm, maybe not, but it is able to tell that the round-tripped `SkyCoord` with velocities is not the same as the original, so it must detect some difference.